lib/cartopy/_crs.pyx

```diff
@@ -264,12 +266,14 @@ cdef class CRS:
             cx *= DEG_TO_RAD
             cy *= DEG_TO_RAD
         status = pj_transform(src_crs.proj4, self.proj4, 1, 1, &cx, &cy, NULL);
-        #if status == -14 or status == -20:
+
+        if trap and status == -14 or status == -20:
```

Comment:
`and` has higher precedence than `or`, so I think you meant to wrap the second two conditions in parentheses.